hazelcast/src/test/java/com/hazelcast/internal/serialization/impl/compact/CompactSerializationTest.java

```diff
@@ -88,6 +105,162 @@ public void testSerializer_withDuplicateFieldNames() {
+        SerializationConfig config = new SerializationConfig();
+
+        assertThatThrownBy(() -> {
+            config.getCompactSerializationConfig()
+                    .setSerializers(new EmployeeDTOSerializer(), new SameTypeNameEmployeeDTOSerializer());
+        }).isInstanceOf(InvalidConfigurationException.class)
+                .hasMessageContaining("Duplicate")
+                .hasMessageContaining("type name");
+    }
+
+    @Test
+    public void testMultipleSerializerRegistration_clearPreviousRegistrations() {
+        SerializationConfig config = new SerializationConfig();
+
+        EmployeeDTOSerializer serializer = spy(new EmployeeDTOSerializer());
+        config.getCompactSerializationConfig()
+                .setEnabled(true)
+                .addSerializer(serializer)
+                .setSerializers(new InnerDTOSerializer(), new MainDTOSerializer());
+
+        SerializationService service = createSerializationService(config);
+        EmployeeDTO employeeDTO = new EmployeeDTO();
+        service.toObject(service.toData(employeeDTO));
+
+        // The previously added serializer should be cleared and not used
+        verify(serializer, never()).read(any());
+        verify(serializer, never()).write(any(), any());
+    }
+
+    @Test
+    public void testMultipleClassRegistration() {
+        SerializationConfig config = new SerializationConfig();
+        config.getCompactSerializationConfig()
+                .setEnabled(true)
+                .setClasses(ExternalizableEmployeeDTO.class, SerializableFoo.class);
+
+        SerializationService service = createSerializationService(config);
+
+        ExternalizableEmployeeDTO externalizableEmployeeDTO = new ExternalizableEmployeeDTO();
+        service.toObject(service.toData(externalizableEmployeeDTO));
+
+        assertFalse(externalizableEmployeeDTO.usedExternalizableSerialization());
+
+        Data data = service.toData(new SerializableFoo());
+        assertTrue(data.isCompact());
+    }
+
+    @Test
+    public void testMultipleClassRegistration_withDuplicateClasses() {
+        SerializationConfig config = new SerializationConfig();
+
+        assertThatThrownBy(() -> {
+            config.getCompactSerializationConfig()
+                    .setClasses(ExternalizableEmployeeDTO.class, ExternalizableEmployeeDTO.class);
+        }).isInstanceOf(InvalidConfigurationException.class)
+                .hasMessageContaining("Duplicate")
+                .hasMessageContaining("type name");
+    }
+
+    @Test
+    public void testMultipleSerializerRegistration_whenThereAreClassRegistrations() {
+        // Make sure that multiple serializer registrations do no override class
+        // registrations
+        SerializationConfig config = new SerializationConfig();
+        config.getCompactSerializationConfig()
+                .setEnabled(true)
+                .setClasses(SerializableFoo.class)
+                .setSerializers(new EmployeeDTOSerializer());
```

Comment:
You said multiple serializer registrations in the test, but do only a single serializer registration here.

Reply:
What I meant with multiple serializer registrations was the method (setSerializers). Let me update the method names to make it more clear